src/vs/workbench/contrib/tasks/browser/task.contribution.ts

```diff
@@ -431,15 +431,24 @@ schema.oneOf = [...(schemaVersion2.oneOf || []), ...(schemaVersion1.oneOf || [])
 const jsonRegistry = <jsonContributionRegistry.IJSONContributionRegistry>Registry.as(jsonContributionRegistry.Extensions.JSONContribution);
 jsonRegistry.registerSchema(tasksSchemaId, schema);
 
-ProblemMatcherRegistry.onMatcherChanged(() => {
-	updateProblemMatchers();
-	jsonRegistry.notifySchemaChanged(tasksSchemaId);
-});
+export class TaskRegistryContribution extends Disposable implements IWorkbenchContribution {
+	static ID = 'taskRegistryContribution';
+	constructor() {
+		super();
+
+		this._register(ProblemMatcherRegistry.onMatcherChanged(() => {
+			updateProblemMatchers();
+			jsonRegistry.notifySchemaChanged(tasksSchemaId);
+		}));
+
+		this._register(TaskDefinitionRegistry.onDefinitionsChanged(() => {
+			updateTaskDefinitions();
+			jsonRegistry.notifySchemaChanged(tasksSchemaId);
+		}));
+	}
+}
+registerWorkbenchContribution2(TaskRegistryContribution.ID, TaskRegistryContribution, WorkbenchPhase.AfterRestored);
```

Comment:
`AfterRestored` is a big change in how this behaves. Do `onMatcherChanged` and `onDefinitionsChanged` only fire after `AfterRestored`?

Reply:
I'm not sure of a precise way to check workbench phase, but when I revert the change and set breakpoints in those event handlers, the window looks like this, restored. 

<img width="1012" alt="Screenshot 2024-06-07 at 9 51 59 AM" src="https://github.com/microsoft/vscode/assets/29464607/526bac6f-61e7-4220-8e26-ed23dbe5abf6">